cattle/compute/__init__.py

```diff
@@ -12,15 +12,7 @@ def _get_handler_category(self, req):
 
     def instance_activate(self, req=None, instanceHostMap=None,
                           processData=None, **kw):
-        instance = instanceHostMap.instance
-        host = instanceHostMap.host
-
-        try:
-            host.clusterConnection = \
-                instanceHostMap.data.fields['clusterConnection']
-            log.info('clusterConnection = %s', host.clusterConnection)
-        except (KeyError, AttributeError):
-            pass
+        instance, host = self.get_instance_host_from_map(instanceHostMap)
```

Comment:
@sonchang putting this note here so we don't forget. As the future enhancement, we should probably move cluster connection information from the instance-host map to host. As this info is not instance specific, and common for the host, it should be loaded once during the host activation.